src/Runner.Worker/OSWarningChecker.cs

```diff
@@ -80,8 +80,70 @@ public async Task CheckOSAsync(IExecutionContext context, IList<OSWarning> osWar
                     context.Global.JobTelemetry.Add(new JobTelemetry() { Type = JobTelemetryType.General, Message = $"An error occurred while checking OS warnings for file '{osWarning.FilePath}' and regex '{osWarning.RegularExpression}': {ex.Message}" });
                 }
             }
+
+            return true;
         }
-    }
 #endif
+
+        private async Task CheckDotNet8CompatibilityAsync(IExecutionContext context, bool omitAnnotation)
+        {
+            try
+            {
+                List<string> output = new();
+                object outputLock = new();
+                using (var process = HostContext.CreateService<IProcessInvoker>())
+                {
+                    process.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs stdout)
+                    {
+                        if (!string.IsNullOrEmpty(stdout.Data))
+                        {
+                            lock (outputLock)
+                            {
+                                output.Add(stdout.Data);
+                                Trace.Info(stdout.Data);
+                            }
+                        }
+                    };
+
+                    process.ErrorDataReceived += delegate (object sender, ProcessDataReceivedEventArgs stderr)
+                    {
+                        if (!string.IsNullOrEmpty(stderr.Data))
+                        {
+                            lock (outputLock)
+                            {
+                                output.Add(stderr.Data);
+                                Trace.Error(stderr.Data);
+                            }
+                        }
+                    };
+
+                    int exitCode = await process.ExecuteAsync(
+                        workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Root),
+                        fileName: Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "testDotNet8Compatibility", $"TestDotNet8Compatibility{IOUtil.ExeExtension}"),
+                        arguments: string.Empty,
+                        environment: null,
+                        cancellationToken: context.CancellationToken);
+
+                    var outputStr = string.Join("\n", output).Trim();
+                    if (exitCode != 0 || !string.Equals(outputStr, "Hello from .NET 8!", StringComparison.Ordinal))
+                    {
+                        if (!omitAnnotation)
+                        {
+                            context.Warning("The runner is not compatible with .NET 8.");
```

Comment:
Have the server send this message, which will solve the following problems:
- Act as a feature flag
- Can update from server to include a link to the changelog, etc.